pymc_marketing/mmm/delayed_saturated_mmm.py

```diff
@@ -452,38 +451,28 @@ def build_model(
 
                 mu_var += control_contributions.sum(axis=-1)
 
-            if (
-                hasattr(self, "fourier_columns")
-                and self.fourier_columns is not None
-                and len(self.fourier_columns) > 0
-                and all(
-                    column in self.preprocessed_data["X"].columns
-                    for column in self.fourier_columns
-                )
-            ):
-                fourier_data_ = pm.Data(
-                    name="fourier_data",
-                    value=self.preprocessed_data["X"][self.fourier_columns],
-                    dims=("date", "fourier_mode"),
-                    mutable=True,
-                )
-                if self.model_config["gamma_fourier"].dims != ("fourier_mode",):
-                    self.model_config["gamma_fourier"].dims = "fourier_mode"
-
-                gamma_fourier = self.model_config["gamma_fourier"].create_variable(
-                    name="gamma_fourier"
+            if self.yearly_seasonality is not None:
+                dayofyear = pm.Data(
+                    name="dayofyear",
+                    values=self.preprocessed_data["X"][
+                        self.date_column
+                    ].dt.dayofyear.to_numpy(),
```

Comment:
Do we check somewhere that the date column indeed has a date type? 🤔

Reply:
Don't think there is a check or cohersion somewhere. It seems like the tvp code assumes it is datetime as well